src/plugins/curator/services/PromotionSubscriber.js

```diff
@@ -195,10 +196,15 @@ export default class PromotionSubscriber extends EventSubscriber {
           const slotNode = slotSchema.createMessage();
           const { name, rendering, widgetRef } = slotData;
 
+          let ref = widgetRef[0];
+          if (typeof ref === 'string') {
+            ref = NodeRef.fromString(widgetRef[0]);
+          }
+
           slotNode
             .set('name', name)
             .set('rendering', rendering.value ? SlotRendering.create(rendering.value) : null)
-            .set('widget_ref', widgetRef[0]);
+            .set('widget_ref', ref);
```

Comment:
@albertgcsula - here's a suggestion:

make the _ref_ a const on Ln 199, and change this line to:

```suggestion
.set('widget_ref', typeof ref === 'string' ? NodeRef.fromString(ref) : ref);
```

Reply:
@arunkarnati okay I made that change for you.